### PR TITLE
Make opendb API more generic

### PR DESCRIPTION
**Pull request by evgeniy-scherbina**
Dependency changes:
- Downgrade `cosmossdk.io/api`: `v0.7.5` -> `v0.3.1`
- Downgrade `cometbft` version declaration: `v0.38.7` -> `v0.37.4`
  - `NOTE`: we're using the fork anyway: `github.com/cometbft/cometbft => github.com/kava-labs/cometbft v0.37.4-kava.1`
  - this change was needed only to make sure that ethermint won't upgrade it's own cometbft dependency following MVS (Minimal Version Selection) algorithm
- Downgrade `golang.org/x/exp`:  `v0.0.0-20240404231335-c0f41cb1a7a0` -> `v0.0.0-20230711153332-06a737ee72cb`

`NOTE`: we downgraded to versions which are used in kava and ethermint

Source code changes:
- allow to configure `db name` instead of hardcoding `application.db`
- allow to configure data directory name instead of hardcoding `data`
- use `db name` as a label in prometheus metrics

Related PR: https://github.com/Kava-Labs/ethermint/pull/63

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -423,77 +428,77 @@ func registerMetrics() {
 }
 
 // report reports metrics to prometheus based on rocksdb props and stats
-func (m *Metrics) report(props *properties, stats *stats) {
+func (m *Metrics) report(dbName string, props *properties, stats *stats) {
```
Comment (karzak) [inline metrics.go:431]: Is there a list/documentation of which metrics are reported? Can we link to it in Confluence?

**Comment by karzak** [comment]
Can you also add a README for this repo @evgeniy-scherbina - can be a separate PR

**Review by karzak — APPROVED** [review]
Changes look fine/safe. I would like to tidy this repo up with a readme that states what its purpose is, as well as document all the rocks db metrics we are collecting and from which databases.

File: metrics.go (hunk 1 above)
Comment (evgeniy-scherbina) [reply metrics.go:431]: Best sources of information:
- grafana dashboard
- source code
- rocksdb wiki

but I'll add a README in a next PR

Comment (evgeniy-scherbina) [reply metrics.go:431]: README is added https://github.com/Kava-Labs/opendb/tree/yevhenii/ethermint?tab=readme-ov-file#purpose-of-this-module

**Pull request merged**